### PR TITLE
[Feature 13 and 12] Added the screens and some of the use case set up in Main have been working. 

### PR DESCRIPTION
**Pull request by xuyangan**
I have merged the progress from feature 12, which deals with creating the GUI and the classes to call it. 
I have made all the screens so, that they call the correct controller through methods that are compiled in a class called Controller Manager. 
The boundaries between the presenter and the screen have also been implemented with significant contributions from Jielin to maintain the clean architecture design. 
We've been mainly working on connecting the front-end features with the back-end features, that is, calling the correct classes.
Part of main has been implemented and still needs some debugging and some other uses to be fully functional, but it is able to run.

### CODE REVIEW

File: src/main/java/entities/QuestionFactoryInterface.java [hunk 1]
```diff
@@ -13,7 +13,17 @@ public interface QuestionFactoryInterface {
...
+        switch (type) {
+            case "MC":
+                return new MultipleChoiceQuestion(questionnaire, variableName, content);
+            case "Scale":
+                return new ScaleQuestion(questionnaire, variableName, content);
+            case "Text":
+                return new TextQuestion(questionnaire, variableName, content);
+        }
+        return null;
+    }
```
Comment (j-y26) [inline src/main/java/entities/QuestionFactoryInterface.java:26]: The interface should not have implementation

**Review by j-y26 — CHANGES_REQUESTED** [review]
This pull request contains multiple regarding the user interface layer of the program and how the user interfaces talks to the back end of the program via the controller and presenters. 

The code follows well clean architecture as the controllers and presenters do not directly depend on the user interface layer. Rather, the user interface implements an interface that is used and defined in the controller/presenter layer and is used to send output data from the presenter to the GUI and supports the manipulation of GUI following specific use case execution. 

More importantly, dependency injection is heavily used by the presenters as the instances of the "display" classes are initiated and injected into the controllers and presenters in the main.

However, a better naming convention should be followed to maintain the consistency between the use case layer and the user interface layer to be clear about how they are related to each other and how we can understand the workflow of which use case manipulates which user interface functionality.

SOLID. The code follows well with the SOLID principle. Despite a large number of functionalities the GUI has to support, each class in the GUI layer is singly responsible for each function and seems to be working in a modular manner -  it is a combination of the classes that together support a single function. Duplicate codes are minimal.

To improve: further testing on the GUI layer and better documentation should be employed.

Some errors exist, which have been noted separately. Please note and revise accordingly.

**Review by j-y26 — APPROVED** [review]
This version of the pull request has addressed most of the issues described in the last review. These changes can be merged intomain.

**Review by Tedward765 — APPROVED** [review]
here is approval

**Review by xustell — APPROVED** [review]
great job.

**Comment by CoqRooster** [comment]
here is approval

**Comment by CoqRooster** [comment]
here is approval

**Pull request merged**